### ISSUE TITLE
twisted.python.test.test_reflect emits DeprecationWarning

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6323]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)| @tomprince reported|
|-|-|
|Trac ID|trac#6323|
|Type|defect|
|Created|2013-02-24 08:17:34Z|
|Branch|https://github.com/twisted/twisted/tree/test_reflect-emits-DeprecationWarning-6323|

`twisted.python.reflect.Accessor` and
`twisted.python.reflect.PropertyAccessor`
are deprecated, but need to be tested.
Mask the deprecation warnings until they can be removed.

<details><summary>Searchable metadata</summary>

```
trac-id__6323 6323
type__defect defect
reporter__tom_prince tom.prince
priority__normal normal
milestone__ 
branch__branches_test_reflect_emits_DeprecationWarning_6323 branches/test_reflect-emits-DeprecationWarning-6323
branch_author__tomprince tomprince
status__closed closed
resolution__fixed fixed
component__core core
keywords__easy easy
time__1361693854000000 1361693854000000
changetime__1362334911000000 1362334911000000
version__None None
owner__tom_prince tom.prince

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_1"></a>|@tomprince commented|
|-|-|

(In [37221]) Branching to test_reflect-emits-DeprecationWarning-6323.

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_3"></a>|@wallrj set owner to @wallrj<br>@wallrj set status to `assigned`|
|-|-|

Reviewing...

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_4"></a>|@wallrj set owner to @tomprince<br>@wallrj set status to `new`|
|-|-|

Code Review

The changes look fine and it was interesting to learn about
warnings.catch_warnings context manager.

But it's not clear to me why this change is necessary.

I ran trial twisted.test.test_reflect against trunk and all tests pass
without any warnings.

The buildbot trunk builds don't show any deprecation warnings:
 * https://buildbot.twistedmatrix.com/builders/py-select-gc/builds/1832/steps/trial/logs/warnings

How do I recreate the problem?

I can see there are a couple of related tickets [#6324](https://github.com/twisted/twisted/issues/6324) and [#6326](https://github.com/twisted/twisted/issues/6326) so I
guess there's a good reason.

Maybe add a short comment to this and the related tickets explaining
the motivation.

Here's what I checked:

 * [Build Results look good](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/test_reflect-emits-DeprecationWarning-6323)
 * Branch merges cleanly to trunk

Please merge if you're sure there's a good reason for the change.

-RichardW.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_5"></a>|@tomprince commented|
|-|-|

Unfortunately, python 2.7 ignores `DeprecationWarning`s by default, so to see them on 2.7, you need to run trial with `python -Walways` or something to see them ([#6348](https://github.com/twisted/twisted/issues/6348)).

You can see them in various 2.6 builders like [this one](https://buildbot.twistedmatrix.com/builders/rhel6-x86_64-py2.6/builds/244/steps/poll/logs/warnings).

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_6"></a>|@tomprince set status to `closed`|
|-|-|

(In [37370]) Merge test_reflect-emits-DeprecationWarning-6323: Filter expected DeprecationWarnings from twisted.python.test.test_reflect.

Author: tom.prince
Reviewers: rwall
Fixes: [#6323](https://github.com/twisted/twisted/issues/6323)

twisted.python.reflect.Accessor and twisted.python.reflect.PropertyAccessor
are deprecated, but need to be tested. Mask the deprecation warnings
until they can be removed.
